crates/ruff_linter/src/rules/pyupgrade/rules/redundant_open_modes.rs

```diff
@@ -72,30 +72,7 @@ pub(crate) fn redundant_open_modes(checker: &mut Checker, call: &ast::ExprCall)
     }
 
     match call.arguments.find_argument("mode", 1) {
-        None => {
-            if !call.arguments.is_empty() {
-                if let Some(keyword) = call.arguments.find_keyword("mode") {
-                    if let Expr::StringLiteral(ast::ExprStringLiteral {
-                        value: mode_param_value,
-                        ..
-                    }) = &keyword.value
-                    {
-                        if let Ok(mode) = OpenMode::from_chars(mode_param_value.chars()) {
-                            let reduced = mode.reduce();
-                            if reduced != mode {
-                                checker.diagnostics.push(create_diagnostic(
-                                    call,
-                                    &keyword.value,
-                                    reduced,
-                                    checker.tokens(),
-                                    checker.stylist(),
-                                ));
-                            }
-                        }
-                    }
-                }
```

Comment:
This block is unreachable.

Reply:
https://github.com/astral-sh/ruff/commit/fd40864924c757ee95d1356c48431358de716d69 updated this block to use `find_argument` but it didn't clean up the `None` arm.

Reply:
Great catch!